source/common/network/tcp_listener_impl.cc

```diff
@@ -82,12 +82,14 @@ void TcpListenerImpl::onSocketEvent(short flags) {
     // Pass the 'v6only' parameter as true if the local_address is an IPv6 address. This has no
     // effect if the socket is a v4 socket, but for v6 sockets this will create an IPv4 remote
     // address if an IPv4 local_address was created from an IPv6 mapped IPv4 address.
-    const Address::InstanceConstSharedPtr& remote_address =
-        (remote_addr.ss_family == AF_UNIX)
-            ? io_handle->peerAddress()
-            : Address::addressFromSockAddr(remote_addr, remote_addr_len,
-                                           local_address->ip()->version() ==
-                                               Address::IpVersion::v6);
+    Address::InstanceConstSharedPtr remote_address;
```

Comment:
Can you put `ASSERT(!isMainThread())` at the top of this function?

Before we allow a conditional RELEASE_ASSERT here we must ensure this is only used on worker threads. If this function is also used on the main thread, then there may be a 'catch' for the throw that's currently in `Address:addressFromSockAddr` and this would cause an invalid address to crash where it was previously handled.

Note that you want such an assert to be outside all these conditions because we want to make sure this function is *only* called in worker threads. IMO it's not sufficient to add an assert that would only trigger if there was a misconfigured address in the main thread.

Reply:
Thanks, will fix.